lex.go

```diff
@@ -100,6 +162,33 @@ func tokenize(reader io.Reader, tokenCh chan NgxToken) {
 			la = "\\" + la
 		}
 
+		if token.Len() > 0 {
+			tokenStr := token.String()
+			if ext, ok := externalLexers[tokenStr]; ok {
+				if nextTokenIsDirective {
+					// saving lex state before emitting tokenStr to know if we encountered start quote
+					lastLexState := lexState
+					if lexState == inQuote {
+						emit(tokenStartLine, true, nil)
+					} else {
+						emit(tokenStartLine, false, nil)
+					}
+
+					externalScanner.tokenLine = tokenLine
+					extTokenCh := ext.Lex(tokenStr)
+					for tok := range extTokenCh {
```

Comment:
Maybe we document on the `Lex` interface method that the implementation must close this channel when it's finished and will not emit any more tokens. Failure to do so and we are stuck here "forever".